_drafts/intro-to-working-with-data-in-CUBA.adoc

```diff
@@ -0,0 +1,201 @@
+
+DataManager is an Abstraction on top of the Java Persistence API (JPA). It hides away certain details that are not required on a day-to-day basis.
+
+It also adds certain capabilities that are not available out of the box in JPA. Those capabilities are mainly the https://doc.cuba-platform.com/manual-6.10/views.html[Views concept] as well as the application of https://doc.cuba-platform.com/manual-6.10/security_subsystem.html[security constrains]. More information about differences between JPA and DataManager can be found in the docs on https://doc.cuba-platform.com/manual-6.10/dm_vs_em.html[DataManager vs. EntityManager].
+
+
+include::includes/petclinic_introduction.adoc[]
+
+
+== Load data with DataManager
+
+DataManager has multiple ways to load data from the database. The following examples contain the most common cases for loading data in different ways.
+
+=== Single Item
+
+The first option is to load a single Entity instance from the database. This option is often required if reference data needs to be read from the database in order to execute some logic for that entity instance or with the help of that entity instance.
+
+In the case of the Petclinic example, there is some business logic which should create a new Visit for a given Pet by its ID.
+
+image:https://dummyimage.com/600x400/00b5ec/ffffff.png&text=Create visit for Pet[] Create visit for Pet overview
+
+
+The part that loads the Pet instance by its ID looks like this with the use of the DataManager API:
+
+
+.VisitServiceBean.java
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+Pet loadPetById(String petId) {
+  return dataManager.load(Pet.class)
+            .query("select e from petclinic$Pet e where e.petId = :petId")
+            .parameter("petId", petId)
+            .one();
+}
+----
+
+DataManager has a https://martinfowler.com/bliki/FluentInterface.html[fluent API] which allows to chain operations on the load operation in order to define parameters to the execution. In this case, the interaction starts with `dataManager.load(Pet.class)`. Then the JPQL query is defined and the associated parameters are configured with the parameter values. The method call `one()` executes the load operation and fetches exactly one item that matches the criteria. If there are no results or multiple results the corresponding exceptions are thrown.
+
+It is up to the application developer to decide if the query can return one or possibly multiple results. In this case it is only possible to create a Pet ID once, we can be sure that the query will return at the most one item.
+
+In this case, the attribute `petId` is not the primary key of the entity. If this would be the case, the query would be even easier to create as the query part is not needed: `dataManager.load(Pet.class).id(petId).one();`
```

Comment:
Please explain here what `petId` actually is.

Reply:
as written above - changed it to identificationNumber - will explain it in there as well.